### PR TITLE
Update PDLs, add brute-force resolving of type sizes

### PR DESCRIPTION
**Pull request by dfrankland**
PDL files have been updated to https://github.com/ChromeDevTools/devtools-protocol/tree/7f780afd2d22bbe5fee4acd9264d1ceec5af6ef9

Since the `BoxStyle` type has been defined after its first use in the `FlexContainerHighlightConfig` type, I've added a small brute-force loop to try to resolve `BoxStyle` (or any other out-of-order types).

`BoxStyle` first use:
https://github.com/ChromeDevTools/devtools-protocol/blob/7f780afd2d22bbe5fee4acd9264d1ceec5af6ef9/pdl/browser_protocol.pdl#L5747-L5765

`BoxStyle` definition:
https://github.com/ChromeDevTools/devtools-protocol/blob/7f780afd2d22bbe5fee4acd9264d1ceec5af6ef9/pdl/browser_protocol.pdl#L5777-L5783

### CODE REVIEW

File: chromiumoxide_cdp/browser_protocol.pdl [hunk 1]
```diff
@@ -5628,6 +5744,50 @@ experimental domain Overlay
...
+        dashed
+        dotted
+
+  # Style information for drawing a box.
+  type BoxStyle extends object
+    properties
+      # The background color for the box (default: transparent)
+      optional DOM.RGBA fillColor
+      # The hatching color for the box (default: transparent)
+      optional DOM.RGBA hatchColor
```
Comment (dfrankland) [inline chromiumoxide_cdp/browser_protocol.pdl:5783]: Here you can see `BoxStyle` out-of-order

File: chromiumoxide_pdl/src/build/generator.rs [hunk 2]
```diff
@@ -213,15 +213,24 @@ impl Generator {
...
+        let mut sequential_retries = 0;
+        while let Some((name, reff)) = refs.pop_front() {
+            match self.type_size.get(&reff).copied() {
+                Some(ref_size) => {
+                    sequential_retries = 0;
+                    self.store_size(&name, Either::Left(ref_size));
+                }
+                None => {
+                    sequential_retries += 1;
+                    if sequential_retries > 10 {
```
Comment (mattsse) [inline chromiumoxide_pdl/src/build/generator.rs:228]: This makes sense to me and ensures that types declared out of order are properly resolved.

Comment (mattsse) [inline chromiumoxide_pdl/src/build/generator.rs:228]: The limit of 10 retries seems arbitrary and possibly unnecessary, comparing against the current length of `refs` is probably more plausible, meaning if the `sequential_retries` is greater than the amount of remaining references in `refs` we tried every ref already and proceed to panic.

Proposing:

```rust
sequential_retries += 1;
if sequential_retries > refs.len() {
panic!(...)
}
```

File: chromiumoxide_pdl/src/build/generator.rs [hunk 3]
```diff
@@ -213,15 +213,24 @@ impl Generator {
             modules.extend(module);
         }
 
-        // fix unresolved type sizes
-        let mut refs = Vec::new();
+        // brute-force fix unresolved type sizes
+        let mut refs = VecDeque::new();
         std::mem::swap(&mut refs, &mut self.ref_sizes);
```
Comment (mattsse) [inline chromiumoxide_pdl/src/build/generator.rs:218]: Now that I'm reading this again I think [`std::mem::replace`](https://doc.rust-lang.org/std/mem/fn.replace.html) would be more idiomatic here.

File: chromiumoxide_pdl/src/build/generator.rs [hunk 4]
```diff
@@ -213,15 +213,24 @@ impl Generator {
...
-        for (name, reff) in refs {
-            let ref_size = *self
-                .type_size
-                .get(&reff)
-                .unwrap_or_else(|| panic!(format!("No type found for ref {}", reff)));
-            self.store_size(&name, Either::Left(ref_size));
+        let mut sequential_retries = 0;
+        while let Some((name, reff)) = refs.pop_front() {
+            match self.type_size.get(&reff).copied() {
```
Comment (mattsse) [inline chromiumoxide_pdl/src/build/generator.rs:221]: Would you rewrite this as `if let`?

```rust
if let Some(ref_size) = self.type_size.get(&reff).copied() {
...
} else {
...
}
```

**Review by mattsse — CHANGES_REQUESTED** [review]
Thanks so much for this!

I have only some minor change requests

File: chromiumoxide_cdp/browser_protocol.pdl (hunk 1 above)
Comment (mattsse) [reply chromiumoxide_cdp/browser_protocol.pdl:5783]: Thanks I wasn't aware that types can be declared after they were used

File: chromiumoxide_pdl/src/build/generator.rs (hunk 2 above)
Comment (dfrankland) [reply chromiumoxide_pdl/src/build/generator.rs:228]: That's a great idea! I definitely chose an arbitrary value so I didn't have to think about it 😅

File: chromiumoxide_pdl/src/build/generator.rs (hunk 3 above)
Comment (dfrankland) [reply chromiumoxide_pdl/src/build/generator.rs:218]: `std::mem::replace` requires that the `src` (second arg) is not a reference, but `self` is `&mut` here, so it's not possible to use I believe.

**Pull request merged**